automation/requirements.txt

```diff
@@ -2,7 +2,7 @@ click~=8.1
 paramiko~=3.4
 semver~=3.0
 requests~=2.31
-boto3>=1.28.0,<1.29.0
+boto3>=1.28.0,<=1.35.16
```

Comment:
s3 functions that directly use boto3:
get, upload, put, stat and listdir.

Also s3fs depends on boto3, so I think It will be the best to run all s3 tests:

integration: `mlrun/tests/integration/aws_s3/test_aws_s3.py`
system tests: `mlrun/tests/system/datastore/test_aws_s3.py`

running spark s3 tests are not required for this change.